machine/state.go

```diff
@@ -25,7 +25,7 @@ const (
 type MachineState struct {
 	ID       string
 	PublicIP string
-	Metadata map[string]string
+	Metadata map[string]string `json:"-"`
```

Comment:
Yes. Since the metadata is no longer inside the main machine object, but rather in independent kv pairs, decoding the machine object could wipe out the metadata.